src/libcore/iter/traits/iterator.rs

```diff
@@ -202,8 +202,9 @@ pub trait Iterator {
     /// Consumes the iterator, counting the number of iterations and returning it.
     ///
     /// This method will evaluate the iterator until its [`next`] returns
-    /// [`None`]. Once [`None`] is encountered, `count()` returns the number of
-    /// times it called [`next`].
+    /// [`None`]. Once [`None`] is encountered, `count()` returns one less than the
```

Comment:
This feels like a confusing way to state things. Perhaps something like this would be better: "`count()` will call [`next()`] until [`None`] is encountered, returning the number of times it saw `Some(_)`."

Reply:
Yeah that's a good point. Just replacing `count()` to "This method" at the beginning feels aesthetically more pleasing. Furthermore using a hyperlink reference to `Some` is probably beneficial.  So may I suggest: "This method will call [`next`] repeatedly until [`None`] is encountered, returning the number of times it saw [`Some`]."

Reply:
Sounds great! <3